modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -725,42 +801,64 @@ namespace cv {
                     std::map<std::string, std::string> &layer_params = i->second;
                     std::string layer_type = layer_params["type"];
 
-                    if (layer_type == "convolutional")
+                    if (layer_type == "convolutional" || layer_type == "connected")
                     {
-                        int kernel_size = getParam<int>(layer_params, "size", -1);
-                        int filters = getParam<int>(layer_params, "filters", -1);
-                        bool use_batch_normalize = getParam<int>(layer_params, "batch_normalize", 0) == 1;
+                        size_t weights_size;
+                        int nums;
+                        bool use_batch_normalize;
+                        cv::Mat weightsBlob;
+                        if(layer_type == "convolutional")
+                        {
+                            int kernel_size = getParam<int>(layer_params, "size", -1);
+                            int filters = getParam<int>(layer_params, "filters", -1);
+                            use_batch_normalize = getParam<int>(layer_params, "batch_normalize", 0) == 1;
+
+                            CV_Assert(kernel_size > 0 && filters > 0);
+                            CV_Assert(current_channels > 0);
+
+                            weights_size = filters * current_channels * kernel_size * kernel_size;
+                            int sizes_weights[] = { filters, current_channels, kernel_size, kernel_size };
+                            weightsBlob.create(4, sizes_weights, CV_32F);
+                            nums = filters;
+                        }
+                        else
+                        {
+                            int output = getParam<int>(layer_params, "output", 1);
+                            use_batch_normalize = getParam<int>(layer_params, "batch_normalize", 0) == 1;
 
-                        CV_Assert(kernel_size > 0 && filters > 0);
-                        CV_Assert(current_channels > 0);
+                            CV_Assert(output>0);
 
-                        size_t const weights_size = filters * current_channels * kernel_size * kernel_size;
-                        int sizes_weights[] = { filters, current_channels, kernel_size, kernel_size };
-                        cv::Mat weightsBlob;
-                        weightsBlob.create(4, sizes_weights, CV_32F);
+                            if(darknet_layers_counter == 0)
+                                current_channels = current_channels * net->height * net->width;
```

Comment:
Looks very tricky. Can we just produce MatShape values?

Reply:
Sure, let me change that.